packages/wrangler/src/pages/functions/buildWorker.ts

```diff
@@ -420,3 +344,79 @@ function blockWorkerJsImports(nodejsCompatMode: NodeJSCompatMode): Plugin {
 		},
 	};
 }
+
+function assetsPlugin(buildOutputDirectory: string | undefined): Plugin {
+	return {
+		name: "Assets",
+		setup(pluginBuild) {
+			const identifiers = new Map<string, string>();
+
+			pluginBuild.onResolve({ filter: /^assets:/ }, async (args) => {
+				const directory = resolve(
+					args.resolveDir,
+					args.path.slice("assets:".length)
+				);
+
+				const exists = await access(directory)
+					.then(() => true)
+					.catch(() => false);
+
+				const isDirectory = exists && (await lstat(directory)).isDirectory();
+
+				if (!isDirectory) {
+					return {
+						errors: [
+							{
+								text: `'${directory}' does not exist or is not a directory.`,
+							},
+						],
+					};
+				}
+
+				// TODO: Consider hashing the contents rather than using a unique identifier every time?
+				identifiers.set(directory, nanoid());
+				if (!buildOutputDirectory) {
+					console.warn(
+						"You're attempting to import static assets as part of your Pages Functions, but have not specified a directory in which to put them. You must use 'wrangler pages dev <directory>' rather than 'wrangler pages dev -- <command>' to import static assets in Functions."
+					);
+				}
+				return { path: directory, namespace: "assets" };
+			});
+
+			pluginBuild.onLoad(
+				{ filter: /.*/, namespace: "assets" },
+				async (args) => {
+					const identifier = identifiers.get(args.path);
+
+					if (buildOutputDirectory) {
+						const staticAssetsOutputDirectory = join(
+							buildOutputDirectory,
+							"cdn-cgi",
+							"pages-plugins",
+							identifier as string
+						);
+						await rm(staticAssetsOutputDirectory, {
+							force: true,
+							recursive: true,
+						});
+						await cp(args.path, staticAssetsOutputDirectory, {
+							force: true,
+							recursive: true,
+						});
+
+						return {
+							// TODO: Watch args.path for changes and re-copy when updated
+							contents: `export const onRequest = ({ request, env, functionPath }) => {
+								const url = new URL(request.url)
```

Comment:
I am a fan of `;`but I know some people aren't.
I doesn't matter so much as long as the code is consistent - it is not in this block

Reply:
I am definitely a fan of `;`... and we should have prettier set up to ensure that.

It makes me worry that these files are not being formatted??

Reply:
Oh of course. This is a string.

Reply:
This string was just moved from being inline and I didn't spend much time looking at the content. :smile: